test/transfer-memory-allocator.test.js

```diff
@@ -0,0 +1,1744 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+
+'use strict';
+
+const assert = require('assert');
+const { TransferMemoryBuffer, TransferMemoryBlock } = require('../lib/transfer-memory-allocator');
+
+describe('transfer-memory-allocator (sync)', function () {
+    it('can create memory allocator (buffer pool)', async function () {
+        const memoryAllocator = new TransferMemoryBuffer(); // use default suggested Buffer size
+        assert.ok(memoryAllocator !== null && memoryAllocator !== undefined);
+
+        assert.ok(memoryAllocator.allocatedBlocks !== null && memoryAllocator.allocatedBlocks !== undefined, "After instantiation, there should be no allocated block but there are");
+        assert.strictEqual(memoryAllocator.allocatedBlocks.length, 0, "After instantiation, allocated blocks length should be 0 but it isn't");
+        assert.strictEqual(memoryAllocator.allocatedBlocks.tail, null);
+        assert.strictEqual(memoryAllocator.allocatedBlocks.head, null);
+    });
+
+    it('does not allow negative buffer pool sizes', async function () {
+        const expectedException = new Error("Transfer Memory Buffer total suggested size must be larger than 0");
+        assert.throws(() => {
+            new TransferMemoryBuffer(-1);
+        }, expectedException);
+    });
+
+    it('does not allow obtaining memory blocks with negative sizes', async function () {
+        const expectedException = new Error("Requested buffer size must be larger than 0");
+        assert.rejects(() => {
+            const memoryAllocator = new TransferMemoryBuffer(10);
+            memoryAllocator.obtainBuffer(-6);
+        }, expectedException);
+    });
+
+    it('does not allow obtaining memory blocks larger than pooled buffer size', async function () {
```

Comment:
looking for a test that tries to create a memory block larger than what memory is available (not the passed in size)